support-lib/jni/djinni_support.cpp

```diff
@@ -585,7 +585,7 @@ void jniSetPendingFromCurrent(JNIEnv * env, const char * ctx) noexcept {
     jniDefaultSetPendingFromCurrent(env, ctx);
 }
 
-void jniDefaultSetPendingFromCurrent(JNIEnv * env, const char * /*ctx*/) noexcept {
+void jniDefaultSetPendingFromCurrentClangCompatibility(JNIEnv * env) {
```

Comment:
Can you give this a name which describes what it is, rather than referring to the clang limitation?  Just `jniDefaultSetPendingFromCurrentImpl` would be fine, or else something decriptive like `jniThrowToSetPending`

That being said, a comment describing why the extra layer is here would be welcome too, though I'd suggest it refer to the general problem (not throwing in a function marked as noexcept) rather than the clang warning itself.